ingest/src/main/java/com/connexta/ingest/client/StoreClient.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.ingest.client;
+
+import com.connexta.ingest.exceptions.StoreException;
+import java.io.IOException;
+import java.net.URI;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.MultipartBodyBuilder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.multipart.MultipartFile;
+
+@Slf4j
+@Service
+public class StoreClient {
+
+  @NotNull private final RestTemplate restTemplate;
+
+  @NotEmpty private final String storeEndpoint;
+
+  public StoreClient(
+      @NotNull final RestTemplate restTemplate,
+      @Qualifier("storeEndpoint") @NotEmpty final String storeEndpoint) {
+    this.restTemplate = restTemplate;
+    this.storeEndpoint = storeEndpoint;
+    log.info("Store URL: {}", storeEndpoint);
+  }
+
+  /** @return the location of the product */
+  public URI store(
+      final Long fileSize, final String mimeType, final MultipartFile file, final String fileName)
+      throws StoreException {
+    final MultipartBodyBuilder builder = new MultipartBodyBuilder();
+    builder.part("fileSize", fileSize);
```

Comment:
❗️ Shouldn't these Strings be part of one JSON blob? We would add that JSON body as a separate part with the `application/json` contentType.

Reply:
Copying my answer from slack:

Good question. I'm not sure it we want to do this the same way as the ingest-api. I'll leave it to you to decide.

Reply:
Yeah, you jogged my memory about the shortcomings of using the OpenAPI generators. We can leave this as is and I'll update the MIS controllers to work like Ingest.

Reply:
I'm fine with leaving it this way. A multipart message is its own beast.